daemon/api.go

```diff
@@ -1316,3 +1323,42 @@ func postCreateUser(c *Command, r *http.Request, user *auth.UserState) Response
 
 	return SyncResponse(createResponseData, nil)
 }
+
+type buyResponseData struct {
+	State string `json:"state,omitempty"`
+}
+
+func postBuy(c *Command, r *http.Request, user *auth.UserState) Response {
+	var buyInstruction store.BuyOptions
+
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&buyInstruction)
+	if err != nil {
+		return BadRequest("cannot decode buying instruction data from request body: %v", err)
+	}
+
+	buyInstruction.Auther, err = c.d.auther(r)
+	if err != nil && err != auth.ErrInvalidAuth {
+		return InternalError("%v", err)
+	}
+
+	s := getStore(c)
+
+	buyResult, err := s.Buy(&buyInstruction)
+
+	switch err {
+	case store.ErrInvalidCredentials:
+		return Unauthorized(err.Error())
+	default:
+		return InternalError("%v", err)
+	case nil:
```

Comment:
Since we're touching that file again anyway, can we please move this case first? Feels slightly more natural than having default in the middle when we actually have another case that is the most expected last.